src/app/service/tool-group/tool-group.service.ts

```diff
@@ -0,0 +1,281 @@
+      .then(() => {
+        return {
+          status: 'success',
+        };
+      })
+      .catch(this.handleError);
+  }
+
+  createOrUpdateRule(
+    toolGroupId: number,
+    ruleId: number,
+    negativeRule: boolean,
+    data: string[] | PraxisData,
+    type: RuleTypeEnum,
+  ) {
+    const isUpdate = ruleId ? true : false;
+    let dataType = '',
+      attributes = {},
+      url = '';
+
+    switch (type) {
+      case RuleTypeEnum.COUNTRY:
+        dataType = 'tool-group-rules-country';
+        attributes = {
+          countries: data,
+          'negative-rule': negativeRule,
+        };
+        url = `${this.toolGroupsUrl}/${toolGroupId}/rules-country`;
+        break;
+      case RuleTypeEnum.LANGUAGE:
+        dataType = 'tool-group-rules-language';
+        attributes = {
+          languages: data,
+          'negative-rule': negativeRule,
+        };
+        url = `${this.toolGroupsUrl}/${toolGroupId}/rules-language`;
+        break;
+      case RuleTypeEnum.PRAXIS:
+        dataType = 'tool-group-rules-praxis';
+        attributes = {
+          ...data,
+          'negative-rule': negativeRule,
+        };
+        url = `${this.toolGroupsUrl}/${toolGroupId}/rules-praxis`;
+        break;
+      default:
+        return;
+    }
+
+    const payload = {
+      data: {
+        type: dataType,
+        attributes,
+      },
+    };
+
+    url += isUpdate ? `/${ruleId}` : '';
+
+    if (isUpdate) {
+      return this.http
+        .put(url, payload, this.authService.getAuthorizationAndOptions())
+        .toPromise()
+        .then((response) => new JsonApiDataStore().sync(response.json()))
+        .catch(this.handleError);
+    }
+    return this.http
+      .post(url, payload, this.authService.getAuthorizationAndOptions())
+      .toPromise()
+      .then((response) => new JsonApiDataStore().sync(response.json()))
+      .catch(this.handleError);
+  }
+
+  deleteRule(toolGroupId: number, ruleId: number, type: RuleTypeEnum) {
+    let url = '';
+    switch (type) {
+      case RuleTypeEnum.COUNTRY:
+        url = `${this.toolGroupsUrl}/${toolGroupId}/rules-country/${ruleId}`;
+        break;
+      case RuleTypeEnum.LANGUAGE:
+        url = `${this.toolGroupsUrl}/${toolGroupId}/rules-language/${ruleId}`;
+        break;
+      case RuleTypeEnum.PRAXIS:
+        url = `${this.toolGroupsUrl}/${toolGroupId}/rules-praxis/${ruleId}`;
+        break;
+      default:
+        return;
+    }
+
+    return this.http
+      .delete(url, this.authService.getAuthorizationAndOptions())
+      .toPromise()
+      .then(() => {
+        return {
+          status: 'success',
+        };
+      })
+      .catch(this.handleError);
+  }
+
+  addOrUpdateTool(
+    toolGroupId: number,
+    id: string,
+    toolID: string,
+    suggestionsWeight: string,
+    isUpdate: boolean,
+  ) {
+    const payload = {
+      data: {
+        type: 'tool-group-tool',
+        attributes: {
+          'suggestions-weight': suggestionsWeight,
```

Comment:
if you hadn't set a `suggestions-weight` in the UI, this gets sent as an empty string which turns into a weight of 0 on the API.

example request:
```json
{
  "data": {
    "type": "tool-group-tool",
    "attributes": {
      "suggestions-weight": ""
    },
    "relationships": {
      "tool": {
        "data": {
          "type": "resource",
          "id": "25"
        }
      }
    }
  }
}
```

Reply:
I'm not sure if we should just omit the `suggestions-weight` attribute in that case, or if you should default to a weight of `1.0`

Reply:
you should send the default weight of `1.0`, the API does not have logic to set a default if there is no weight provided